packages/flutter_tools/lib/src/linux/build_linux.dart

```diff
@@ -12,13 +12,22 @@ import '../convert.dart';
 import '../globals.dart';
 import '../project.dart';
 
-/// Builds the Linux project through the project shell script.
+/// Builds the Linux project through the Makefile.
 Future<void> buildLinux(LinuxProject linuxProject, BuildInfo buildInfo) async {
+  /// Cache flutter root in linux directory.
+  linuxProject.editableHostAppDirectory.childFile('.generated_flutter_root')
+    ..createSync(recursive: true)
+    ..writeAsStringSync(Cache.flutterRoot);
+
+  final String buildFlag = buildInfo?.isDebug == true ? 'debug' : 'release';
```

Comment:
Why the `== true` on this and the next line? The value is already a boolean, right?

Reply:
Guarded against nulls. I saw a few in tests, I need to shake it out separately.